app/routes.py

```diff
@@ -1 +1,268 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, jsonify, abort, make_response, request
+from datetime import date
+from app import db
+import requests
+import os
+
+from app.models.task import Task
+from app.models.goal import Goal
+
+# - - - - - - - - - - - - - - - - - #
+###~~~---- helper function ----~~~###
+# - - - - - - - - - - - - - - - - - #
+
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response({"details": f"Invalid data"}, 400))
+
+    model = cls.query.get(model_id)
+
+    if not model:
+        abort(make_response({"message": f"Sorry, {cls.__name__} {model_id} does not exist"}, 404))
+
+    return model
+
+# - - - - - - - - - - - - - - - - - #
+###~~~--- task model routes ---~~~###
+# - - - - - - - - - - - - - - - - - #
+
+task_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+@task_bp.route("", methods=['POST'])
+
+# define a route for creating a task
+def create_task():
+    request_body = request.get_json()
+
+    try:
+        new_task = Task(
+            title = request_body["title"],
+            description = request_body["description"],
+        )
+    except:
+        abort(make_response({"details": f"Invalid data"}, 400))
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {"task": new_task.to_dict()}, 201
+
+# define a route for getting all tasks
+@task_bp.route("", methods=["GET"])
+def read_all_tasks():
+    # querries
+    title_query = request.args.get("title")
+    description_query = request.args.get("description")
+    completed_at_query = request.args.get("completed_at")
+    sort_query = request.args.get("sort")
+
+    if title_query:
+        tasks = Task.query.filter_by(title=title_query)
+    
+    elif description_query:
+        tasks = Task.query.filter_by(description=description_query)
+    
+    elif completed_at_query:
+        tasks = Task.query.filter_by(completed_at=completed_at_query)
+    
+    elif sort_query:
+        if sort_query == 'asc':
+            tasks = Task.query.order_by(Task.title).all()
+        elif sort_query == 'desc':
+            tasks = Task.query.order_by(Task.title.desc()).all()
```

Comment:
Nice work on getting the database to do the heavy lifting when it came to sorting the tasks!